src/components/tl/ucp/allgather/allgather_knomial.c

```diff
@@ -30,6 +30,23 @@
  * a new virtual rank number - "vrank".
  * As such allgather must keep to this ranking to be aligned with scatter.
  */
+static ucc_rank_t calc_recv_dist(ucc_rank_t team_size, ucc_rank_t rank,
```

Comment:
very similar to calc_recv_dist in allgather_knomial.c
maybe they should be a joint function in some h file that they both read from to remove code duplication?